pkg/server/providers.go

```diff
@@ -58,7 +70,11 @@ func (s *Server) registerProviders() error {
 				continue
 			}
 
-			err = s.ProviderManager.RegisterProvider(pluginPath)
+			if strings.HasSuffix(pluginPath, manager.INITIAL_SETUP_LOCK_FILE_NAME) {
```

Comment:
This can be check up top (`if file.IsDir() && file.Name() != manager.INITIAL_SETUP_LOCK_FILE_NAME`)

Reply:
This wouldn't work since what we are "searching" for here are the directories (e.g. the `docker-provider` dir) and the function above (`getPluginPaths`) is the one checking the actual files inside.

I improved the variable naming to make this a bit more clear and moved the lock file check inside `getPluginPaths`